src/streamlink/plugins/afreeca.py

```diff
@@ -4,55 +4,66 @@
 from streamlink.plugin import Plugin, PluginArgument, PluginArguments
 from streamlink.plugin.api import validate
 from streamlink.stream import HLSStream
+from streamlink.stream.hls_filtered import FilteredHLSStreamReader, FilteredHLSStreamWriter
 
 log = logging.getLogger(__name__)
 
-CHANNEL_API_URL = "http://live.afreecatv.com:8057/afreeca/player_live_api.php"
-STREAM_INFO_URLS = "{rmd}/broad_stream_assign.html"
 
-CHANNEL_RESULT_ERROR = 0
-CHANNEL_RESULT_OK = 1
+class AfreecaHLSStreamWriter(FilteredHLSStreamWriter):
+    def should_filter_sequence(self, sequence):
+        return "preloading" in sequence.segment.uri
 
-QUALITYS = ["original", "hd", "sd"]
 
-QUALITY_WEIGHTS = {
-    "original": 1080,
-    "hd": 720,
-    "sd": 480
-}
+class AfreecaHLSStreamReader(FilteredHLSStreamReader):
+    __writer__ = AfreecaHLSStreamWriter
 
-_url_re = re.compile(r"https?://play\.afreecatv\.com/(?P<username>\w+)(?:/\d+)?")
 
-_channel_schema = validate.Schema(
-    {
-        "CHANNEL": {
-            "RESULT": validate.transform(int),
-            validate.optional("BPWD"): validate.text,
-            validate.optional("BNO"): validate.text,
-            validate.optional("RMD"): validate.text,
-            validate.optional("AID"): validate.text,
-            validate.optional("CDN"): validate.text
-        }
-    },
-    validate.get("CHANNEL")
-)
-
-_stream_schema = validate.Schema(
-    {
-        validate.optional("view_url"): validate.url(
-            scheme=validate.any("rtmp", "http")
-        ),
-        "stream_status": validate.text
-    }
-)
+class AfreecaHLSStream(HLSStream):
+    def open(self):
+        reader = AfreecaHLSStreamReader(self)
+        reader.open()
+        return reader
 
 
 class AfreecaTV(Plugin):
-    login_url = "https://member.afreecatv.com:8111/login/LoginAction.php"
+    _re_bno = re.compile(r"var nBroadNo = (?P<bno>\d+);")
+    _re_url = re.compile(r"https?://play\.afreecatv\.com/(?P<username>\w+)(?:/(?P<bno>:\d+))?")
+
+    CHANNEL_API_URL = "http://live.afreecatv.com/afreeca/player_live_api.php"
+    CHANNEL_RESULT_OK = 1
+    QUALITYS = ["original", "hd", "sd"]
+    QUALITY_WEIGHTS = {
+        "original": 1080,
+        "hd": 720,
+        "sd": 480,
+    }
+
+    _schema_channel = validate.Schema(
+        {
+            "CHANNEL": {
+                "RESULT": validate.transform(int),
+                validate.optional("BPWD"): validate.text,
+                validate.optional("BNO"): validate.text,
+                validate.optional("RMD"): validate.text,
+                validate.optional("AID"): validate.text,
+                validate.optional("CDN"): validate.text
```

Comment:
I know you've only moved it from the module's global scope to the plugin class, but you could change `validate.text` to `str` in one go here (down below as well).